TODO/Android-Studio-Tips.md

```diff
@@ -1,832 +1,542 @@
+- **更多：**如果一个语句已经补全，当你执行该操作时，则会直接跳到下一行，即使光标不在当前行的行末。
 
 ![](https://lh6.googleusercontent.com/-oZeWSimrvoU/VAWr5QoA-oI/AAAAAAAANQE/0LxL0LkN8Jw/w281-h124-no/16-completestatement.gif)
 
+# 10. 条件断点（Conditional Breakpoints）
 
-# Conditional Breakpoints
-
-In a nutshell, only do break when a certain condition is met. You can enter any java expression that returns a boolean based on the current scope. 
-And do enjoy the fact that the condition textbox supports code completion!
-
-Shortcut: 
-  - **Right click on a breakpoint and enter a condition**.﻿
+- **描述：**简单说，就是当设定的条件满足时，才会触发断点。你可以基于当前范围输入一个java布尔表达式，并且条件输入框内是支持代码补全的。
+- **调用：**右键需要填写表达式的断点，然后输入布尔表达式。
 
 ![](https://lh6.googleusercontent.com/-p9k6JiNLQmY/VBAweflrkYI/AAAAAAAANX8/gCaufjGbd1c/w514-h264-no/22-conditionalbreakpoint.gif)
 
+# 11. 上下文信息（Context Info）
 
-# Context Info
-
-
-So this will show you where you are when your scope definition is out of the scrolling area. Usually, this will be the name of the class or inner class  but it might also be the current method name.
-
-Its better use, IMHO, is to get a quick look at what the current class extends or implements.
-
-It also works in xml files.
-
-Shortcut: **Alt+Q**
+- **描述：**当前区域超过滚动区域，执行该操作将显示所在的上下文信息，通常它显示的是类名或者内部类类名或者当前所在的方法名。该操作在xml文件中同样适用。
+- **调用：**Menu → View → Context Info
+- **快捷键：**Alt + Q (Windows/Linux)
+- **更多：**比较好用的是，可以用来快速看当前类继承的父类或实现的接口。
 
 ![](https://lh4.googleusercontent.com/-FNg2h15F4c0/VD-rJupXgkI/AAAAAAAAOL4/lfaQmbjwpaw/w574-h174-no/47-contextinfo.gif)
 
+# 12. 删除行（Delete Line）
 
-# Delete Line
-
-It deletes the current line without having to select it.
-If a block is already selected, it will delete every line in the selection.
-
-Shortcut: **cmd+backspace(mac)** or **ctrl+y(windows/linux)﻿**
+- **描述：**如果没选中，则删除光标所在行，如果选中，则会删除选中所在的所有行。
+- **快捷键：**Cmd + Delete(OS X)、Ctrl + Y(Windows/Linux)
 
 ![](https://lh3.googleusercontent.com/-bP5WOVMfp7A/U_cpQi0bvhI/AAAAAAAAM9c/dcvvJu1US40/w265-h103-no/10-deleteline.gif)
 
+# 13. 断点失效（Disable Breakpoints）
 
-# Disable Breakpoints
-
-This will disable the breakpoint. Particularly useful when you have some complicated conditional or logging breakpoint that you don't need right now but don't want to recreate next time.
-
-Shortcut: 
-  - Mouse: **Alt+LeftClick** on an existing breakpoint in the left gutter
-  - There is no keyboard shortcut but you can create one if you use it often enough.﻿
+- 这个操作将使得断点失效。当处在一种复杂的状态或者某个断点你现在不需要，但是下次又不用重新创建，该操作是很方便的。
+- **调用：**按住Alt，然后单击断点即可。
 
 ![](https://lh3.googleusercontent.com/-hNk0kuL1WBM/VBbQXamG8-I/AAAAAAAANeM/ynfSJ5hqCvA/w365-h235-no/25-diablebreakpoint.gif)
 
+# 14. 行复制（Duplicate Line）
 
-# Duplicate Line
-
-Just as it says: it will copy the current line and paste it below without interfering with your clipboard. It can be pretty useful when it's used with the move line shortcut!
-
-Shortcut: **cmd+d(mac)** or **ctrl+d(windows/linux)﻿**
+- **描述：**复制当前行，并粘贴到下一行。
```

Comment:
It can be pretty useful when it's used with the move line shortcut!

这个命令配合移动行快捷键非常有用。

实际上这是我在eclipse时代就用的非常多的快捷键组合了，确实很好用～eclipse里面是ctrl-alt-下箭头，移动行是alt-方向键～